wordpress-mobile/WordPress-Android
# Implement Gravatar QuickEditor

This PR implements the QuickEditor from the Gravatar SDK. It fully replaces the Avatar update flow, including uploading already existing photos or new ones from the camera.

**A few things worth noting:**
- One of Gravatar's compose dependencies bumps the foundation library. This caused the WP build to fail because a few things got deprecated. These issues were fixed in https://github.com/wordpress-mobile/WordPress-Android/commit/5b1adac3c25f43908889e1c99c3603e47a6db8d1. If you would prefer, I can update the compose BOM (only this specific artifact) and apply those changes in a separate PR.
- Gravatar uses UCrop like the WP app. We found an issue where the cropped image size (file size) is not changed. We've tried to fix that in UCrop ([PR](https://github.com/Yalantis/uCrop/pull/926)) but things aren't moving fast there, so we released the Automattic fork with the fix applied. I had to update WP to the Automattic version to not cause conflicts. 
- I've added a feature flag for the QuickEditor. Its purpose is a safety mechanism to be able to turn off the integration in case something doesn't work as it should. I'm not sure where to set this remotely or whether this has to happen now or when it's needed. Can you guide me here? 

Here's how the QuickEditor looks like in WP on the `MeFragment`

https://github.com/user-attachments/assets/f4488cf1-76d9-4e2b-815b-37dbb2528a64

And here's the `SignupEpilogueFragment`

https://github.com/user-attachments/assets/a6d5fabf-8651-47ae-a47d-16e7f8c23ec3


-----

## To Test:



1. Go through the sign-up flow
2. On the `SignupEpilogueFragment` tap the avatar to upload a new image
3. Confirm the avatar was uploaded
4. Continue and go to the `MeFragment` by tapping the `Me` option
5. Tap on the avatar
6. Confirm the QuickEditor was opened
7. Upload a new avatar, play with the QE.
8. Modify the `GRAVATAR_QUICK_EDITOR` feature flag in build.gradle to `false`
9. Repeat steps 1-7 - this should behave same as it works now without the QuickEditor.

-----

## Regression Notes

1. Potential unintended areas of impact

    - no

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
